File: ndet_tran.py

def ndet_tran(detections=None,num=None,*args,**kwargs):
    ## divide the detection matrix into cells according to the frame number
    global params
    for flag in range(num):
        det[flag]=detections[detections[:][0] == flag][:]
        det[flag][:][0:2]=[]
        # line below
        det[flag][:][2:4] = det[flag][:][0:2] + det[flag][:][2:4]
        det[flag][:][0:4] = floor(det[flag][:][0:4])
        det[flag][:][5:]=[]

    detections = det
    return detections

if __name__ == '__main__':
    ndet_tran() #Test using random variables

File: nonconverted/ndet_tran.m
function detections = ndet_tran(detections,num)
%% divide the detection matrix into cells according to the frame number
global params;
for flag= 1:num
    det{flag} = detections(detections(:,1) == flag,:);
    det{flag}(:,1:2) = [];
    
    % if your detection format is [x1,y1,x2,y2] not [x,y,w,h], comment the
    % line below
    det{flag}(:,3:4) = det{flag}(:,1:2)+det{flag}(:,3:4);    
    det{flag}(:,1:4) = floor(det{flag}(:,1:4));
    det{flag}(:,6:end)=[];
end
detections = det;
end